### DISCUSSION TITLE
Can I use other services besides Azure?

### DISCUSSION THREAD

**Discussion by DC-Sebastian** [openai-php/client#216] [Q&A] [answered]
Can I also use other providers that use the same api structure as open api. With Azure this is already possible but what about e.g. https://docs.endpoints.anyscale.com/ which output the data in the same format as open ai?
Can I create a factory for this, or change the Endpoint url or is this only possible for Azure?

**Comment by Anton-V-K** [comment]
This client seems to support the customization of the base url (according to the documentation):
```php
$yourApiKey = 'YOUR_ANYSCALE_ENDPOINT_API_KEY';

$client = OpenAI::factory()
    ->withApiKey($yourApiKey)
    ->withBaseUri('https://api.endpoints.anyscale.com/v1') // default: api.openai.com/v1
    ->make();
```
Have you tried this approach?

**Reply by DC-Sebastian** [reply, marked as answer]
In the meantime it works, at the time I wrote my question there was a problem with a missing header which led to an error message, but this was fixed a few months ago.
